### DISCUSSION TITLE
Published builds not starting with library version past 7.23.1

### DISCUSSION THREAD

**Discussion by nevaran** [dlemstra/Magick.NET#929] [Help]
For a while now I had to stay with MagickNET 7.23.1, because of a weird issue where the application wont start at all in a published build with any of the later versions of the library.
It does not give any crash reports or anything, just will not start.

Im using WPF with .Net 5.

The only thing I can remember that starts with the program always is the `CacheDirectory `change I do in the `App.xaml.cs` file.

**Comment by dlemstra** [comment]
Is it possible to create a super small project that demonstrates the issue? I don't have enough information now to investigate this.

**Reply by nevaran** [reply]
I dont exactly have a modular code so it might take quite some time to strip it away, but my actual project is public if you need to create a build?

[heres the source code](https://github.com/nevaran/Fast-Image-Viewer-Standalone)

Using the library is all done in the `FIVStandard.Core.Tools` class from what I can remember, but since the later versions dont even get to that code, the only place that I use it, is as I mentioned before, in the App.xaml.cs where i change the cache dir

The project uses ffmpeg but it should work without the actual binaries, just needs the MagickNET library to be updated.

**Reply by nevaran** [reply]
I think i figured out why it wouldnt start?
In publish builds you guys generate a `Notice.txt` file that I havent included in the installer's list of files.
Im guessing this is a thing added after 7.23.1?

**Reply by dlemstra** [reply]
That was added but is not required for your program to run. They just state which external libraries are being used. 

I cannot help you debug your whole program. Maybe you are missing some files when you publish the program? But there were no new binaries added. Maybe your code breaks somewhere else? Why do you think the upgrade breaks it?

**Reply by nevaran** [reply]
I have separatedly taken the exe (whole program) and magick library file in a separate folder and tried running it, and it does not run unless that notice text file is also added in there.

I also managed to also strip a good chunk of the project but there is still quite a lot of there. Still, here it is: [github](https://github.com/nevaran/FIV-Lite)

Here is the published folder and attempting to run with and without the text file generated: [google drive](https://drive.google.com/file/d/14vE0Dpz52LWH9C4luYYA6j6R2CnGhgFh/view?usp=sharing)

Edit: stripped much more of the project, now it is basically bare bones. I have also tried by moving the **cache directory** change to another class or completely removing it; no change

**Reply by nevaran** [reply]
Do you need me to do anything else?

**Reply by dlemstra** [reply]
Does the program run correctly if the notice.txt file is included?

**Reply by nevaran** [reply]
Yup, perfectly fine

**Reply by nevaran** [reply]
Any news on the issue?

**Reply by dlemstra** [reply]
This should be resolved in the latest release.

**Reply by nevaran** [reply]
Got it, thanks
